### ISSUE TITLE
Cannot translate the value of keypath 'auth.failed'. Use the value of keypath as default.

### ISSUE THREAD

**Issue by Morinohtar** [kirschbaum-development/laravel-translations-loader#9]
## Packages
```json
"vue": "^2.6.6",
"vue-i18n": "^8.8.2",
"@kirschbaum-development/laravel-translations-loader": "^0.4.2",
```

I'm doing a simple test:

### app.js has the following code among other things

```js
window.Vue = require('vue');

import VueI18n from 'vue-i18n';

import languageBundle from '@kirschbaum-development/laravel-translations-loader/php?parameters={{ $1 }}!@kirschbaum-development/laravel-translations-loader';

Vue.use(VueI18n);

const fallbackLocale = 'en';

const i18n = new VueI18n({
    fallbackLocale,
    locale: 'en',
    messages: languageBundle
});
```

### ExampleComponent.vue has the following code among other things

```vue
{{ $t('auth.failed') }}
```

This will produce the error: 
`Cannot translate the value of keypath 'auth.failed'. Use the value of keypath as default.`

But if i do this:

```js
const i18n = new VueI18n({
    fallbackLocale,
    locale: 'en',
    messages: {
        en: {
            auth: {
                failed: 'For real?!'
            }
        },
    }
});
```

It works... so, any ideas why it is not liking the `languageBundle`? Anything missing or wrong?

**Comment by luisdalmolin** [comment]
Are you using a default Laravel project? What's the location of your `app.js` file? Also, can you `console.log` the `languageBundle` and post here?

**Comment by Morinohtar** [comment]
The `app.js` is located in another folder `resources/js/backend/administrators/app.js`. Do i need something extra to compensate the difference?

**Comment by luisdalmolin** [comment]
Yeah, that's definitely the problem. What node version are you using? I'm going to try to replicate this here. 

Also, what is the laravel mix / webpack version you are using?

**Comment by Morinohtar** [comment]
node: 11.3.0
npm: 6.4.1
laravel-mix: 4.0.14
webpack: 4.29.6

**Comment by luisdalmolin** [comment]
@Morinohtar any chance you could push the failing code to a repo so I can take a look?

**Comment by Morinohtar** [comment]
Sorry, i cannot, it is a project from the company where i work.

But i can post the content of the crucial files:

### bootstrap.js 
#### located in: resources/js/backend/administrators

```js
window._ = require('lodash');

/**
 * We'll load jQuery and the Bootstrap jQuery plugin which provides support
 * for JavaScript based Bootstrap features such as modals and tabs. This
 * code may be modified to fit the specific needs of your application.
 */

try {
    window.Popper = require('popper.js').default;
    window.$ = window.jQuery = require('jquery');

    require('bootstrap');
} catch (e) {}

/**
 * We'll load the axios HTTP library which allows us to easily issue requests
 * to our Laravel back-end. This library automatically handles sending the
 * CSRF token as a header based on the value of the "XSRF" token cookie.
 */

window.axios = require('axios');

window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';

const locale = document.head.querySelector('meta[name="locale"]');
const localeRegional = document.head.querySelector('meta[name="locale-regional"]');

window.locale = locale.content;
window.localeRegional = localeRegional.content;

/**
 * Next we will register the CSRF Token as a common header with Axios and jQuery so that
 * all outgoing HTTP requests automatically have it attached. This is just
 * a simple convenience so we don't have to attach every token manually.
 */

let csrfToken = document.head.querySelector('meta[name="csrf-token"]');

if (csrfToken) {
    window.axios.defaults.headers.common['X-CSRF-TOKEN'] = csrfToken.content;

    window.$.ajaxSetup({headers: {'X-CSRF-TOKEN': csrfToken.content}});
} else {
    console.error('CSRF token not found: https://laravel.com/docs/csrf#csrf-x-csrf-token');
}

```

### app.js 
#### located in: resources/js/backend/administrators

```js
require('./bootstrap');

// Moment
window.moment = require('moment');
moment.locale(window.locale);

// Activate tooltip of bootstrap.
window.$('[data-toggle="tooltip"]').tooltip({html: true});

// WebFont
import WebFont from 'webfontloader';
WebFont.load({
    google: {"families": ["Montserrat:400,500,600,700", "Noto+Sans:400,700"]},
    active: function () {
        sessionStorage.fonts = true;
    }
});

// niceScroll
window.niceScroll = require('jquery.nicescroll');

// SweetAlert2
import Swal from 'sweetalert2';
window.Swal = window.swal = Swal;

// Noty
import Noty from 'noty';
window.Noty = window.noty = Noty;

// Vue.js
window.Vue = require('vue');

// Vue SweetAlert2
import VueSweetalert2 from 'vue-sweetalert2';
Vue.use(VueSweetalert2);

// VueI18n
import VueI18n from 'vue-i18n';
import languageBundle from '@kirschbaum-development/laravel-translations-loader/php?parameters={{ $1 }}!@kirschbaum-development/laravel-translations-loader';
Vue.use(VueI18n);

const fallbackLocale = 'en';

const i18n = new VueI18n({
    fallbackLocale,
    locale: window.locale,
    messages: languageBundle
});

Vue.component('example-component', require('./components/ExampleComponent.vue').default);

const app = new Vue({
    el: '#app',
    i18n
});
```

### ExampleComponent.vue

```vue
<template>
    <div>
        {{ $t('auth.failed') }}
        <br>
        {{ message }}
    </div>
</template>

<script>
    export default {
        data() {
            return {
                message: 'For real?!'
            }
        }
    };
</script>
```

### mix script
```js
mix.js('resources/js/backend/administrators/app.js', 'public/js/backend/administrators/app.min.js')
    .extract()
    .sass(
        'resources/sass/backend/administrators/app.scss',
        'public/css/backend/administrators/app.min.css'
    )
    .options({
        fileLoaderDirs: {
            images: 'assets/images/backend/administrators',
            fonts: 'assets/fonts/backend/administrators'
        }
    });
```

**Comment by luisdalmolin** [comment]
Hmm, I thought it could be related to the vendor extraction, but is not (at least on my local). The only other reason I can see is if you are using a different location for language files or your node environment has some different configurations.

I pushed this repo with what I'm using to test: https://github.com/luisdalmolin/translations-loader-issue-9

Can you make a test using this repo and see what you get in `console.log`.

**Comment by Morinohtar** [comment]
I didn't change the location of the locales, it's in the same place.
Hmmm... maybe node... going to try something else.
I will try that repo as soon as i can.

**Comment by Morinohtar** [comment]
Ok, the issue is with `npm`. I ran it inside the VM (homestead) and it works.

I'm not that an expert on `npm`, so do you have any idea what config might be messing this?

**Comment by luisdalmolin** [comment]
@Morinohtar can you run `npm config ls -l` and post the output here?

**Comment by Morinohtar** [comment]
This is the `config` of the `host`:

```sh
$ npm config ls -l
; cli configs
long = true
metrics-registry = "https://registry.npmjs.org/"
scope = ""
user-agent = "npm/6.9.0 node/v11.3.0 win32 x64"

; builtin config undefined
prefix = "C:\\Users\\Xpto\\AppData\\Roaming\\npm"

; default values
access = null
allow-same-version = false
also = null
always-auth = false
audit = true
audit-level = "low"
auth-type = "legacy"
before = null
bin-links = true
browser = null
ca = null
cache = "C:\\Users\\Xpto\\AppData\\Roaming\\npm-cache"
cache-lock-retries = 10
cache-lock-stale = 60000
cache-lock-wait = 10000
cache-max = null
cache-min = 10
cafile = undefined
cert = null
cidr = null
color = true
commit-hooks = true
depth = null
description = true
dev = false
dry-run = false
editor = "notepad.exe"
engine-strict = false
fetch-retries = 2
fetch-retry-factor = 10
fetch-retry-maxtimeout = 60000
fetch-retry-mintimeout = 10000
force = false
git = "git"
git-tag-version = true
global = false
global-style = false
globalconfig = "C:\\Users\\Xpto\\AppData\\Roaming\\npm\\etc\\npmrc"
globalignorefile = "C:\\Users\\Xpto\\AppData\\Roaming\\npm\\etc\\npmignore"
group = 0
ham-it-up = false
heading = "npm"
https-proxy = null
if-present = false
ignore-prepublish = false
ignore-scripts = false
init-author-email = ""
init-author-name = ""
init-author-url = ""
init-license = "ISC"
init-module = "C:\\Users\\Xpto\\.npm-init.js"
init-version = "1.0.0"
json = false
key = null
legacy-bundling = false
link = false
local-address = undefined
loglevel = "notice"
logs-max = 10
; long = false (overridden)
maxsockets = 50
message = "%s"
; metrics-registry = null (overridden)
node-options = null
node-version = "11.3.0"
noproxy = null
offline = false
onload-script = null
only = null
optional = true
otp = null
package-lock = true
package-lock-only = false
parseable = false
prefer-offline = false
prefer-online = false
; prefix = "C:\\Program Files\\nodejs" (overridden)
preid = ""
production = false
progress = true
proxy = null
read-only = false
rebuild-bundle = true
registry = "https://registry.npmjs.org/"
rollback = true
save = true
save-bundle = false
save-dev = false
save-exact = false
save-optional = false
save-prefix = "^"
save-prod = false
scope = ""
script-shell = null
scripts-prepend-node-path = "warn-only"
searchexclude = null
searchlimit = 20
searchopts = ""
searchstaleness = 900
send-metrics = false
shell = "C:\\WINDOWS\\system32\\cmd.exe"
shrinkwrap = true
sign-git-commit = false
sign-git-tag = false
sso-poll-frequency = 500
sso-type = "oauth"
strict-ssl = true
tag = "latest"
tag-version-prefix = "v"
timing = false
tmp = "C:\\Users\\Xpto\\AppData\\Local\\Temp"
umask = 0
unicode = false
unsafe-perm = true
update-notifier = true
usage = false
user = 0
; user-agent = "npm/{npm-version} node/{node-version} {platform} {arch}" (overridden)
userconfig = "C:\\Users\\Xpto\\.npmrc"
version = false
versions = false
viewer = "browser"


```

**Comment by luisdalmolin** [comment]
Ok, I'm going to try to replicate this in a windows environment over the weekend. In the meantime, using the instructions noted here (https://github.com/kirschbaum-development/laravel-translations-loader#using-a-different-folder) under the `Using a different folder` here should probably solve your issue.

Basically you need:

1 - Create an `index.js` file in the `resources/lang` directory.
2 - Change your import to: `import languageBundle from '@kirschbaum-development/laravel-translations-loader/php?parameters={{ $1 }}!@kirschbaum-development/laravel-translations-loader!./../lang/index.js';`

Let me know if that works.

**Comment by Morinohtar** [comment]
That solution worked ;)

**Comment by luisdalmolin** [comment]
@Morinohtar The problem was the path separators on windows. That is fixed on version 0.4.4 if you want to upgrade.
